src/libraries/System.Private.CoreLib/src/System/IO/File.cs

```diff
@@ -379,7 +379,15 @@ public static void WriteAllBytes(string path, byte[] bytes)
             fs.Write(bytes, 0, bytes.Length);
 #else
             using SafeFileHandle sfh = OpenHandle(path, FileMode.Create, FileAccess.Write, FileShare.Read);
-            RandomAccess.WriteAtOffset(sfh, bytes, 0);
+
+            long offset = 0;
+            Span<byte> buffer = bytes;
+            do
+            {
+                int bytesWritten = RandomAccess.WriteAtOffset(sfh, buffer, offset);
```

Comment:
Should this somehow handler `!buffer.IsEmpty && bytesWritten == 0` to prevent infinite loop? (Logically, it would make more sense to me as a `while` loop instead of `do-while` loop).